ph5/utilities/validation.py

```diff
@@ -128,70 +134,48 @@ def check_response_info(info, ph5, unique_filenames_n_i, checked_data_files,
+    header = ("array {0}, station {1}, channel {2}, "
+              "response_table_n_i {3}: ").format(info['array'],
+                                                 info['sta'],
+                                                 info['cha_id'],
+                                                 info['n_i'])
     if Response_t is None:
-        errmsg = "No response entry for n_i=%s." % info['n_i']
-        addError(errmsg, errors, logger)
-        return False
+        errmsg = ("%sresponse_t has no entry for n_i=%s"
+                  % (header, info['n_i']))
+        return False, errmsg
     if info['n_i'] == -1:
         # metadata no response signal
-        return False
+        errmsg = ("%sMetadata response with n_i=-1 has no response data."
+                  % header)
+        return False, errmsg
 
-    m_file = None
+    d_file = None
+    s_file = ''
     # check resp file from metadata
-    m_ret = check_resp_file_name(
-        Response_t, info, 'metadata', unique_filenames_n_i, errors, logger)
-    if m_ret is not None:
-        m_file, m_path = m_ret
-        if m_path is not None:
-            # resp file is metadata type => don't need to check other types
-            if m_file not in checked_data_files.keys():
-                if check_resp_data(ph5.ph5, m_file, errors, logger):
-                    checked_data_files[m_file] = True
-                    return (m_path, None)
-                else:
-                    checked_data_files[m_file] = False
-                    return False
-            if checked_data_files[m_file]:
-                return (m_path, None)
-            else:
-                return False
-
-    # check sensor
-    s_ret = check_resp_file_name(
-        Response_t, info, 'sensor', unique_filenames_n_i, errors, logger)
-    if s_ret is not None:
-        s_file, s_path = s_ret
-        if s_file not in checked_data_files.keys():
-            if check_resp_data(ph5.ph5, s_file, errors, logger):
-                checked_data_files[s_file] = True
-            else:
-                checked_data_files[s_file] = False
-                s_path = None
-        else:
-            if not checked_data_files[s_file]:
-                s_path = None
-    else:
-        s_file, s_path = None, None
-
-    # check das
-    ret = check_resp_file_name(
-        Response_t, info, 'das', unique_filenames_n_i, errors, logger, m_file)
-    if ret is not None:
-        d_file, d_path = ret
-        if d_file not in checked_data_files.keys():
-            if check_resp_data(ph5.ph5, d_file, errors, logger):
-                checked_data_files[d_file] = True
-                return (d_path, s_path)
-            else:
-                checked_data_files[d_file] = False
-                return (None, s_path)
-        else:
-            if checked_data_files[d_file]:
-                return (d_path, s_path)
-            else:
-                return (None, s_path)
-    else:
-        return (None, s_path)
```

Comment:
This logic seems to be different. Is there a way to test the new logical path and verify that it is working as expected?

Reply:
I had to fix this to meet your requirement of changes.
Before, response data were checked only when the file names in the response table were consistent with information in array table and the name return is the one created from the information in array table.
Now, any inconsistencies between response table and information in array table will be logged as warning and not interrupt the data checking. Then data will be checked and if it is passed, the response file names in the response table will be returned.

Reply:
As a clarification, are ph5_validate.log files still output if there are inconsistencies between response table and information in array table?

What does a warning look like? Can you provide an example?